### DISCUSSION TITLE
unable to playback / show mkv video

### DISCUSSION THREAD

**Discussion by LiveG24** [rejetto/hfs#759] [Q&A] [answered]
Hello
first, thanks to mighty rejetto for the hfs server ;-)

second, i have a folder with around 8000 mkv video of 5 min / 70-80mb each
hfs works very good in downloading the files but open / show did not work for me
( btw, what is the difference between open and show ? )

I did try to modify the mime options ( video/mp4 or video/webem ) but no luck :-/

Any hint ?

Thanks

**Comment by rejetto** [comment]
hey!
HFS just asks the browser to play the file, but this works only for files supported by the **browser**. If it doesn't play, 99% is because the browser doesn't support it.
File extension is not the only variable, because inside a mkv you can have different codecs, and only some are supported.
Mime can't help, you are just making things worse :)

I decided to inaugurate the FAQ page
https://github.com/rejetto/hfs/wiki/FAQ#q-what-is-the-difference-between-open-and-show

**Reply by LiveG24** [reply]
got it !
btw I forgot to mention the codec in the mkv ... :-/

<img width="507" alt="Screenshot 2024-10-01 at 10 44 37" src="https://github.com/user-attachments/assets/6a8a10dc-7160-40e1-a2ae-d2b8a6ee73bc">

is possible to disable the open and show in the file menu? 
so only download will be available

**Comment by rejetto** [comment]
use this style
`#menu-entry-open, #menu-entry-show { display: none }`

**Comment by LiveG24** [comment]
wonderful ! thanks a lot

**Comment by LiveG24** [comment]
> HFS just asks the browser to play the file, but this works only for files supported by the **browser**. If it doesn't play, 99% is because the browser doesn't support it. File extension is not the only variable, because inside a mkv you can have different codecs, and only some are supported. Mime can't help, you are just making things worse :)
> 

I have tried to open the file with chrome ( same browser i am using toward hfs ) and the mkv play fine.
I am puzzled now :-/

**Comment by rejetto** [comment]
you can record a video of the problem, or give me access to your hfs (a@rejetto.com)

**Reply by LiveG24** [reply]
i've sent you the details ;-)

**Comment by SanokKule** [comment]
Mkv container is not really supported by the browsers, rather they support webm which is based on mkv.
https://en.wikipedia.org/wiki/WebM
While chrome with right mime will play most formats in mkv (any combination of av1/vp9/vp8/h265/h264 video and opus/aac/vorbis/flac/mp3 worked for me), firefox will only play av1/vp9/vp8 for video and opus/vorbis for audio in mkv. If any other format is present, it won't play.
If you want mkv to play in most browsers, limit yourself to av1/vp8/vp9 and vorbis/opus and use this mime config 
![image](https://github.com/user-attachments/assets/39b83922-f1e6-4f6d-b920-a4a1718b5562)
I'm not sure about safari so can't comment there.

**Reply by LiveG24** [reply]
thank you for the hint
but yesterday we made some test and discovered that if I open the same mkv directly with chrome, it works fine 👍 
if in hfs I shift+click over the file, it works fine as well 👍 

it don't work only if I click over the file in hfs and then on the file menu I choose "show" :-/
and btw this only happen on newer chrome 129, because on old 109 or 128 chrome it works correctly
and only with my own mkv file ... maybe the small size (70-80 mb) play a role in this issue

**Comment by rejetto** [comment]
i confirm that the mime trick solves SOME problem: "open" on my chrome128 on mac, works only with the mime trick. I'm testing using the file that you gave me, "2024-10-01 00-02-44.mkv".
without the mime trick, "open" will just download.
anyway, "show" works on my mac even without.

when i switch to my windows with chrome129, instead, "open" works without the trick, while "show" fails 🫠
i change mime, and "open" now DOESN'T work anymore, and neither "show", so it made just worse.
this is quite confusing.

i can't say if the main difference is windows/mac or version 128/129.
i suspect it's the OS, anyway, because chrome may "inherit" some codec from it.

**Comment by rejetto** [comment]
@LiveG24  i continued with my tests, and maybe i understood something more:
when you autoplay with the shortcut, it does play correctly, and the problem is only when the video is open WITHOUT playing.
HFS receives an error and decides that the file is not playable. That's not true, tho.
I guess that when the video has never started, it tries to get some picture to show, and THAT fails.
Of course this is type of error should not be fatal, but hfs is not distinguishing.
you can try this test version, just right-click on the check-for-updates in admin/home
[hfs-mac-arm64-0.54.0-beta5.2.zip](https://github.com/user-attachments/files/17266932/hfs-mac-arm64-0.54.0-beta5.2.zip)
[hfs-windows-x64-0.54.0-beta5.2.zip](https://github.com/user-attachments/files/17266933/hfs-windows-x64-0.54.0-beta5.2.zip)

**Comment by rejetto** [comment]
the same file works on my Windows 11 with Chrome 129
https://github.com/user-attachments/assets/d7554876-d293-4189-b808-85f4f66e5672

**Comment by LiveG24** [comment]
sorry for the error, i did download the wrong update ...

I can confirm that on W10 the beta 5.2 works fine with both click + show or shift+click !

thanks a lot 🖖

**Comment by rejetto** [comment, marked as answer]
[hfs-mac-x64-0.54.0-beta5.22.zip](https://github.com/user-attachments/files/17278416/hfs-mac-x64-0.54.0-beta5.22.zip)

daje! here is correct "preview" binary for you

**Comment by LiveG24** [comment]
thanks again !
